core/controllers/reader.py

```diff
@@ -1000,6 +1001,30 @@ def get(self, exploration_id):
         })
         self.render_json(self.values)
 
+    @acl_decorators.can_play_exploration
+    def put(self, _):
+        """Handles PUT requests."""
```

Comment:
It seems like this doesn't even have the same signature as the GET handler, so I don't think we should bundle these into one handler. Could you make a separate handler just for this -- e.g. StoryProgressHandler (or perhaps reinstate StoryNodeCompletionHandler, though I think the new name I suggested is better)?

Reply:
Done, reinstated StoryProgressHandler.